src/main/java/randoop/generation/GrtStringFuzzer.java

```diff
@@ -0,0 +1,259 @@
+   * </ul>
+   */
+  private enum StringFuzzingOperation {
+    /** The operation to insert a character at a random index. */
+    INSERT {
+      @Override
+      Sequence inputs(int length) {
+        int idx = Randomness.nextRandomInt(length + 1);
+        char c = (char) (Randomness.nextRandomInt(PRINTABLE_ASCII_SPAN) + PRINTABLE_ASCII_START);
+        return Sequence.concatenate(
+            Sequence.createSequenceForPrimitive(idx), Sequence.createSequenceForPrimitive(c));
+      }
+    },
+    /** The operation to remove a character at a random index. */
+    REMOVE {
+      @Override
+      Sequence inputs(int length) {
+        return Sequence.createSequenceForPrimitive(Randomness.nextRandomInt(length));
+      }
+    },
+    /** The operation to replace a character at a random index with a random character. */
+    REPLACE {
+      @Override
+      Sequence inputs(int length) {
+        int i1 = Randomness.nextRandomInt(length);
+        int i2 = Randomness.nextRandomInt(length);
+        int start = Math.min(i1, i2), end = Math.max(i1, i2);
+        String r =
+            String.valueOf(
+                (char) (Randomness.nextRandomInt(PRINTABLE_ASCII_SPAN) + PRINTABLE_ASCII_START));
+        return Sequence.concatenate(
+            Sequence.createSequenceForPrimitive(start),
+            Sequence.createSequenceForPrimitive(end),
+            Sequence.createSequenceForPrimitive(r));
+      }
+    },
+    /** The operation to extract a substring from a random start index to a random end index. */
+    SUBSTRING {
+      @Override
+      Sequence inputs(int length) {
+        int i1 = Randomness.nextRandomInt(length);
+        int i2 = Randomness.nextRandomInt(length);
+        int start = Math.min(i1, i2), end = Math.max(i1, i2);
+        return Sequence.concatenate(
+            Sequence.createSequenceForPrimitive(start), Sequence.createSequenceForPrimitive(end));
+      }
+    };
+
+    /** The set of all StringFuzzingOperation values. */
+    private static final StringFuzzingOperation[] VALUES = values();
+
+    /**
+     * Return a random StringFuzzingOperation.
+     *
+     * @return a random StringFuzzingOperation
+     */
+    static StringFuzzingOperation random() {
+      return VALUES[Randomness.nextRandomInt(VALUES.length)];
+    }
+
+    /** The set of all StringBuilder methods for this operation. */
+    private static final Map<StringFuzzingOperation, List<Executable>> METHOD_CACHE;
+
+    static {
+      EnumMap<StringFuzzingOperation, List<Executable>> m =
+          new EnumMap<>(StringFuzzingOperation.class);
+      m.put(INSERT, init("insert", int.class, char.class));
+      m.put(REMOVE, init("deleteCharAt", int.class));
+      m.put(REPLACE, init("replace", int.class, int.class, String.class));
+      m.put(SUBSTRING, init("substring", int.class, int.class));
+      METHOD_CACHE = Collections.unmodifiableMap(m);
+    }
+
+    /** Return the list of methods for this operation. */
```

Comment:
More specifically, is it a list of methods such that running them all in sequence performs the given operation?  Please clarify.